save-core/src/jvmTest/kotlin/org/cqfn/save/core/integration/GeneralTest.kt

```diff
@@ -0,0 +1,88 @@
+package org.cqfn.save.core.integration
+
+import org.cqfn.save.core.files.readFile
+import org.cqfn.save.core.result.Pass
+import org.cqfn.save.core.utils.CurrentOs
+import org.cqfn.save.core.utils.ProcessBuilder
+import org.cqfn.save.core.utils.getCurrentOs
+import org.cqfn.save.core.utils.isCurrentOsWindows
+import org.cqfn.save.reporter.Report
+
+import okio.FileSystem
+import okio.Path.Companion.toPath
+
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+@Suppress(
+    "TOO_LONG_FUNCTION",
+    "INLINE_CLASS_CAN_BE_USED",
+    "MISSING_KDOC_TOP_LEVEL",
+    "MISSING_KDOC_CLASS_ELEMENTS")
+@OptIn(ExperimentalSerializationApi::class)
+class GeneralTest {
+    private val fs = FileSystem.SYSTEM
+
+    @Test
+    fun `examples test`() {
+        val binDir = "../save-cli/build/bin/" + when (getCurrentOs()) {
+            CurrentOs.LINUX -> "linuxX64"
+            CurrentOs.MACOS -> "macosX64"
+            CurrentOs.WINDOWS -> "mingwX64"
+            else -> return
+        } + "/debugExecutable"
+
+        assertTrue(fs.exists(binDir.toPath()))
+
+        val saveExecutableFiles = fs.list(binDir.toPath()).filter { fs.metadata(it).isRegularFile }
+        // Binary should be created at this moment
+        assertTrue(saveExecutableFiles.isNotEmpty())
+
+        val examplesDir = "../examples/kotlin-diktat/"
+
+        val actualSaveBinary = saveExecutableFiles.last()
+        val saveBinName = if (isCurrentOsWindows()) "save.exe" else "save"
```

Comment:
Isn't it `.kexe` on Linux and mac?

Reply:
I need `.exe` on Windows just to show this OS, that current file is executable, on Mac and Linux it doesn't matter in general, it is ELF file, for which we give permissions for execution by `chmod`